rs/tests/nns/node_removal_from_registry_test.rs

```diff
@@ -116,23 +116,19 @@ pub fn test(env: TestEnv) {
         )
         .await;
         vote_execute_proposal_assert_executed(&governance_canister, proposal_id).await;
-        // Confirm that the node was indeed removed by sending the proposal again and asserting failure.
-        let proposal_id = submit_external_proposal_with_test_id(
-            &governance_canister,
-            NnsFunction::RemoveNodes,
-            RemoveNodesPayload {
-                node_ids: vec![unassigned_node_id],
-            },
-        )
-        .await;
-        vote_execute_proposal_assert_failed(
-            &governance_canister,
-            proposal_id,
-            format!(
-                "Aborting node removal: Node Id {} not found in the registry",
-                unassigned_node_id
-            ),
-        )
-        .await;
+
+        // Confirm that the node was indeed removed by checking the unassigned node list in the registry.
+        topology
+            .block_for_newer_registry_version()
+            .await
+            .expect("Could not obtain updated registry.");
+        let topology = env.topology_snapshot();
+        assert_eq!(
+            topology
+                .unassigned_nodes()
+                .find(|node| node.node_id == unassigned_node_id)
+                .map(|node| node.node_id),
```

Comment:
Not sure why map here? Is this just to avoid spam? Especially when tests fail, I think it makes sense to err on the side of over-communicating.

In the same vein, filter is maybe better than find, because filter would comprehensively fetch ALL the bad data, not just the first example.

This is a nonblocking suggestion.

Reply:
We need a map because `&IcNodeSnapshot` does not have Debug implemented, so we can't assert_eq! it. And frankly... node_id is less verbose and is all we need anyway.

Reply:
I changed to `.filter` instead of `.find` as you requested, although we should really break if there is _any_ matching node id. But I can see how having the complete list might be useful in debugging/logs.